src/Components/bookHub/BookHub.js

```diff
@@ -73,35 +81,104 @@ function CardsDisplay() {
 
   const searchByTerm = async () => {
     let matched = [];
-    setSearchTerms(searchTerms.toLowerCase());
-    if (defaultSearch) {
+    if (category === 'title' || category === 'description' || category === 'identity') {
       // title/description/identity
       matched = allBooks.filter(isMatchTitleDescIdentity);
     } else {
-      // we can figure out how to filter both fields at once later
-      let res = await Filter('Creator', 'authored');
-      matched.push(...res);
-      res = await Filter('Creator', 'illustrated');
-      matched.push(...res);
+      let res;
+      if (category === 'author') {
+        res = await SearchFilter('Creator', 'authored');
+        matched.push(...res);
+      }
+      if (category === 'illustrator') {
+        res = await SearchFilter('Creator', 'illustrated');
+        matched.push(...res);
+      }
 
       matched = [...new Set(matched)];
     }
 
     setFilteredBooks(matched);
   };
 
+  /* We will separate useEffect for *hopefully* clearer code */
+  // Grab books from Airtable if nothing is in local state
+  useEffect(() => { if (!allBooks.length) { getCards(); } }, [allBooks]);
+
+  // Search function
   useEffect(() => {
-    if (!allBooks.length) { getCards(); }
-    if (searchTerms) {
-      (async () => searchByTerm())();
-    } else {
-      setFilteredBooks(allBooks);
+    if (alignment === 'Search') {
+      if (searchTerms) {
+        (async () => searchByTerm())();
+      } else {
+        setFilteredBooks(allBooks);
+      }
     }
-  }, [allBooks, searchTerms, defaultSearch]);
+  }, [allBooks, searchTerms, category, alignment]);
+
+  // Filter function
+  useEffect(() => {
+    if (alignment === 'Filter') {
+      const validGradeTags = gradeRangeMetadata.slice(
+        gradeRangeMetadata.indexOf(rangeInput.grade.min),
+        gradeRangeMetadata.indexOf(rangeInput.grade.max) + 1,
+      );
+      const validAgeTags = ageRangeMetadata.slice(
+        ageRangeMetadata.indexOf(rangeInput.age.min),
+        ageRangeMetadata.indexOf(rangeInput.age.max) + 1,
+      );
+
+      setFilteredBooks(allBooks.filter(
+        (record) => {
+          const incomingGradeTags = gradeRangeMetadata.slice(
+            gradeRangeMetadata.indexOf(record.fields.grade_min),
+            gradeRangeMetadata.indexOf(record.fields.grade_max) + 1,
+          );
+          const incomingAgeTags = ageRangeMetadata.slice(
+            ageRangeMetadata.indexOf(`${record.fields.age_min}`),
+            ageRangeMetadata.indexOf(`${record.fields.age_max}`) + 1,
+          );
+
+          return ((incomingAgeTags.some((val) => validAgeTags.indexOf(val) !== -1)
+      && incomingGradeTags.some((value) => validGradeTags.indexOf(value) !== -1))
+      && (multiSelectInput['race/ethnicity'].length === 0
```

Comment:
I think we can make a sub function here because of how similar the following sections of code are to clean this code up. The code below is all repeats of the same function with just the field changed.

Reply:
I think dan changed the logic for this